app/models/compute_attribute.rb

```diff
@@ -32,8 +32,12 @@ def respond_to_missing?(method, include_private = false)
     vm_attrs.has_key?(method.to_s) || super
   end
 
+  def validate_vm_attrs
+    compute_resource.validate_vm_attrs(vm_attrs)
+  end
+
   def normalized_vm_attrs
-    compute_resource.normalize_vm_attrs(vm_attrs)
+    compute_resource.normalize_vm_attrs(validate_vm_attrs)
```

Comment:
Are you sure it's needed here? this method is only used when displaying the attributes in the api

Reply:
I am pretty sure, I will explain:
when creating the compute attributes in the API this method is used too and if the attributes are not valid they are saved anyway which is what I was trying to avoid.

Reply:
The only place i found where this method is called is https://github.com/theforeman/foreman/blob/develop/app/views/api/v2/compute_attributes/base.json.rabl#L5, which is indeed called when rendering an api response for `create`, but that call only occurs after the object is saved successfully and the attributes have already been normalized and serialized. If the save fails I think it does not render but rather an error response is rendered.